runtime/include/gpu/cuda/chpl-gpu-gen-includes.h

```diff
@@ -106,6 +106,14 @@ __device__ __host__ static inline void chpl_gpu_printTimeDelta(
   printf("%s%u\n", msg, end - start);
 }
 
+__device__ static inline void chpl_gpu_force_sync() {
+  asm volatile("bar.sync 0;" : : : "memory");
```

Comment:
Seeing this here made me wonder whether we could just call `__syncthreads()` here. Even if it does fix, I wouldn't trust it too much. It is more of a curiosity at this point.

Should we put a link to the bug report and the solution here, as well?

Not that we have many different GPUs we test this on, but I hope this solution is portable. At least not less portable than our runtime today.

Reply:
I've tried just using `__syncthreads()` here and it fails as well (makes sense since I think inlining occurs before other clang optimizations).